components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRPCOpsImpl.kt

```diff
@@ -20,71 +24,166 @@ import net.corda.lifecycle.LifecycleCoordinatorFactory
             }
+            is StopEvent -> coordinator.updateStatus(LifecycleStatus.DOWN)
+            is RegistrationStatusChangeEvent -> {
+                when (event.status) {
+                    LifecycleStatus.ERROR -> {
+                        coordinator.closeManagedResources(setOf(CONFIG_HANDLE))
+                        coordinator.postEvent(StopEvent(errored = true))
+                    }
+                    LifecycleStatus.UP -> {
+                        // Receive updates to the RPC and Messaging config
+                        coordinator.createManagedResource(CONFIG_HANDLE) {
+                            configurationReadService.registerComponentForUpdates(
+                                coordinator,
+                                requiredKeys
+                            )
+                        }
+                    }
+                    else -> logger.debug { "Unexpected status: ${event.status}" }
+                }
+                coordinator.updateStatus(event.status)
+                logger.info("${this::javaClass.name} is now ${event.status}")
+            }
+            is ConfigChangedEvent -> {
+                if (requiredKeys.all { it in event.config.keys } and event.keys.any { it in requiredKeys }) {
+                    val rpcConfig = event.config.getConfig(ConfigKeys.RPC_CONFIG)
+                    val messagingConfig = event.config.getConfig(ConfigKeys.MESSAGING_CONFIG)
+                    val duration = Duration.ofMillis(rpcConfig.getInt(ConfigKeys.RPC_ENDPOINT_TIMEOUT_MILLIS).toLong())
+                    // Make sender unavailable while we're updating
+                    coordinator.updateStatus(LifecycleStatus.DOWN)
+                    coordinator.createManagedResource(SENDER) {
+                        virtualNodeSenderFactory.createSender(duration, messagingConfig)
+                    }
+                    coordinator.updateStatus(LifecycleStatus.UP)
+                }
+            }
         }
     }
 
+    /**
+     * Sends the [request] to the configuration management topic on bus.
+     *
+     * @property request is a [VirtualNodeManagementRequest]. This an enveloper around the intended request
+     * @throws CordaRuntimeException If the message could not be published.
+     * @return [VirtualNodeManagementResponse] which is an envelope around the actual response.
+     *  This response corresponds to the [VirtualNodeManagementRequest] received by the function
+     * @see VirtualNodeManagementRequest
+     * @see VirtualNodeManagementResponse
+     */
+    private fun sendAndReceive(request: VirtualNodeManagementRequest): VirtualNodeManagementResponse {
+        if (!isRunning) throw IllegalStateException(
+            "${this.javaClass.simpleName} is not running! Its status is: ${lifecycleCoordinator.status}"
+        )
+
+        return lifecycleCoordinator.getManagedResource<VirtualNodeSender>(SENDER)!!.sendAndReceive(request)
```

Comment:
I wonder if we should be a bit more defensive here and check if `lifecycleCoordinator.getManagedResource<VirtualNodeSender>(SENDER)` is `null`.
Like we do in `VirtualNodeMaintenanceRPCOpsImpl#sendAndReceive`.

```suggestion
        val sender = lifecycleCoordinator.getManagedResource<VirtualNodeSender>(SENDER)
            ?: throw IllegalStateException("Sender not initialized, check component status for ${this.javaClass.name}")

        return sender.sendAndReceive(request)
```